generators/add-events/index.js

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const path = require('path')
+const Generator = require('yeoman-generator')
+
+const { atLeastOne } = require('../../lib/utils')
+
+const { eventCodes } = require('../../lib/constants')
+
+// we have one actions generator per service, an action generator could generate different types of actions
+const eventCodeToActionGenerator = {
+  [eventCodes.cloudEvents]: path.join(__dirname, 'cloud-events/index.js')
+}
+
+const eventCodeToTitle = {
+  [eventCodes.cloudEvents]: 'Publish Cloud Events to Adobe I/O'
+}
+
+/*
+      'initializing',
+      'prompting',
+      'configuring',
+      'default',
+      'writing',
+      'conflicts',
+      'install',
+      'end'
+      */
+
+class AddEvents extends Generator {
+  constructor (args, opts) {
+    super(args, opts)
+
+    // options are inputs from CLI or yeoman parent generator
+    this.option('skip-prompt', { default: false })
+    this.option('adobe-services', { type: String, default: '' })
+    this.option('skip-install', { type: String, default: false })
+
+    // todo throw meaningful error if add actions in a non existing project, but what defines a project?
+  }
+
+  async prompting () {
+    const eventCodesList = [eventCodes.cloudEvents]
```

Comment:
here you can just define a list that directly points to a choice i.e.
`eventGeneratorChoices = [{ name: 'Publish Cloud Events to Adobe I/O', value: path.join(__dirname, 'cloud-events/index.js') }]`